File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/ViewModelsInteraction/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations

Namespace MVVMDemo.ViewModelsInteraction

    Public Class MainViewModel
        Inherits ViewModelBase

        <Command>
        Public Sub ShowDetail(ByVal item As String)
            Dim service As IDialogService = ServiceContainer.GetService(Of IDialogService)()
            service.ShowDialog(dialogButtons:=MessageButton.OK, title:="Detail", documentType:=Nothing, parameter:=item, parentViewModel:=Me)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/ViewModelsInteraction/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;

namespace MVVMDemo.ViewModelsInteraction {
    public class MainViewModel : ViewModelBase {
        [Command]
        public void ShowDetail(string item) {
            IDialogService service = ServiceContainer.GetService<IDialogService>();
            service.ShowDialog(
                dialogButtons: MessageButton.OK,
                title: "Detail", 
                documentType: null, 
                parameter: item, 
                parentViewModel: this);
        }
    }
}
